src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/PushHookProcessor.java

```diff
@@ -158,11 +158,14 @@ public String getSourceName() {
                                 return Collections.emptyMap();
                             }
                             Map<SCMHead, SCMRevision> result = new HashMap<>();
-                            for (BitbucketPushEvent.Change change: getPayload().getChanges()) {
+                            for (final BitbucketPushEvent.Change change: getPayload().getChanges()) {
                                 if (change.isClosed()) {
                                     result.put(new BranchSCMHead(change.getOld().getName(), type), null);
                                 } else {
-                                    BranchSCMHead head = new BranchSCMHead(change.getNew().getName(), type);
+                                    SCMHead head = "tag".equalsIgnoreCase(change.getNew().getType()) ?
+                                        //TODO: from where to retrieve timestamp ?
```

Comment:
?

Reply:
The 2nd parameter of `BitbucketTagSCMHead` required a timestamp milliseconds. This information is currently not available in `BitbucketPushEvent` nor in `BitbucketPushEvent.Change` so currently, I'm using `Instant.now().getMillis()` which is approximatively (~100 ms depending on the connection) correct

Reply:
You can not use the milliseconds of current time. The reason because my PR was in hold is because should be changed also the DTO that rapresents the bitbucket Cloud event to get the correct timestamp of commit